odxtools/comparamspec.py

```diff
@@ -51,7 +52,11 @@ def _finalize_init(self, database: "Database", odxlinks: OdxLinkDatabase) -> Non
         super()._finalize_init(database, odxlinks)
 
     def _resolve_snrefs(self, context: SnRefContext) -> None:
+        context.comparam_spec = self
+
         super()._resolve_snrefs(context)
 
         for ps in self.prot_stacks:
             ps._resolve_snrefs(context)
+
+        context.comparam_spec = None
```

Comment:
What is this?

Reply:
this is to make it consistent with the diag layers, i.e., to make it possible for SNREFs within comparam specs to be resolved relative to the comparam spec (fortunately, this is currently not used, but I think it is a bit strange if diag layers handle this differently than comparam specs...)

Reply:
done: #401